bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,181 @@
+}
+
+
+class Hangman(commands.Cog):
+    """
+    Cog for the Hangman game.
+
+    Hangman is a classic game where the user tries to guess a word, with a limited amount of tries.
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command()
+    async def hangman(
+            self,
+            ctx: commands.Context,
+            min_length: int = 0,
+            max_length: int = 25,
+            min_unique_letters: int = 0,
+            max_unique_letters: int = 25,
+            singleplayer: Literal["s", "m", "S", "M"] = "s",
+    ) -> None:
+        """
+        Play hangman against the bot, where you have to guess the word it has provided!
+
+        The arguments for this command mean:
+        - min_length: the minimum length you want the word to be (i.e. 2)
+        - max_length: the maximum length you want the word to be (i.e. 5)
+        - min_unique_letters: the minimum unique letters you want the word to have (i.e. 4)
+        - max_unique_letters: the maximum unique letters you want the word to have (i.e. 7)
+        - singleplayer: writing 's' means you want to play by yourself, and only you can suggest letters,
+            - writing 'm' means you want multiple players to join in and guess the word.
+        """
+        # Changing singleplayer to a boolean
+        singleplayer = True if singleplayer.lower() == 's' else False
+
+        # Filtering the list of all words depending on the configuration
+        filtered_words = [
+            word for word in ALL_WORDS
+            if min_length < len(word) < max_length
+            and min_unique_letters < len(set(word)) < max_unique_letters
+        ]
+
+        if not filtered_words:
+            filter_not_found_embed = Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No words could be found that fit all filters specified.",
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=filter_not_found_embed)
+            return
+
+        word = choice(filtered_words)
+
+        user_guess = "_" * len(word)
+        tries = 6
+        guessed_letters = set()
+
+        # Checking if the game is singleplayer
+        def check(msg: Message) -> bool:
+            if singleplayer:
+                return msg.author == ctx.author
+            else:
+                # Multiplayer mode
+                return msg.author != self.bot
+
+        hangman_embed = Embed(
+            title="Hangman",
+            color=Colours.python_blue,
+        )
+        hangman_embed.set_image(url=IMAGES[tries])
+        hangman_embed.add_field(
+            name=f"You've guessed `{user_guess}` so far.",
+            value="Guess the word by sending a message with the letter!",
+            inline=False,
+        )
+        hangman_embed.set_footer(text=f"Tries: {tries}")
+        original_message = await ctx.send(embed=hangman_embed)
```

Comment:
This code is duplicated at the end of the while loop.

You can remove this duplication if you move this whole block into the while loop and have it be the first thing that happens.